File: php/tests/Unit/Domain/Catalogue/ValueObject/ItemUomConversionTest.php
<?php
declare(strict_types=1);

namespace Tests\Unit\Domain\Catalogue\ValueObject;

use Alxarafe\App\Domain\Catalogue\ValueObject\ItemUomConversion;
use Alxarafe\App\Domain\Catalogue\ValueObject\UomId;
use InvalidArgumentException;
use PHPUnit\Framework\TestCase;

final class ItemUomConversionTest extends TestCase
{
    private UomId $fromUomId;
    private UomId $toUomId;

    protected function setUp(): void
    {
        $this->fromUomId = new UomId('018e4e3a-3e7b-7b3e-8000-000000000001');
        $this->toUomId = new UomId('018e4e3a-3e7b-7b3e-8000-000000000002');
    }

    public function testCreateConversion(): void
    {
        $conversion = new ItemUomConversion($this->fromUomId, $this->toUomId, 2.0);
        self::assertTrue($this->fromUomId->equals($conversion->fromUomId()));
        self::assertTrue($this->toUomId->equals($conversion->toUomId()));
        self::assertSame(2.0, $conversion->factor());
    }

    public function testNegativeFactor(): void
    {
        $this->expectException(InvalidArgumentException::class);
        new ItemUomConversion($this->fromUomId, $this->toUomId, -1.0);
    }

    public function testZeroFactor(): void
    {
        $this->expectException(InvalidArgumentException::class);
        new ItemUomConversion($this->fromUomId, $this->toUomId, 0.0);
    }

    public function testSameUom(): void
    {
        $this->expectException(InvalidArgumentException::class);
        new ItemUomConversion($this->fromUomId, $this->fromUomId, 1.0);
    }

    public function testEquals(): void
    {
        $a = new ItemUomConversion($this->fromUomId, $this->toUomId, 2.0);
        $b = new ItemUomConversion($this->fromUomId, $this->toUomId, 2.0);
        self::assertTrue($a->equals($b));
    }
}

File: java/src/test/java/com/alxarafe/domain/catalogue/valueobject/ItemUomConversionTest.java
package com.alxarafe.domain.catalogue.valueobject;

import com.alxarafe.app.domain.catalogue.valueobject.ItemUomConversion;
import com.alxarafe.app.domain.catalogue.valueobject.UomId;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class ItemUomConversionTest {

    private final UomId fromUomId = new UomId("018e4e3a-3e7b-7b3e-8000-000000000001");
    private final UomId toUomId = new UomId("018e4e3a-3e7b-7b3e-8000-000000000002");

    @Test
    void createConversion() {
        var conversion = new ItemUomConversion(fromUomId, toUomId, 2.0);
        assertEquals(fromUomId, conversion.fromUomId());
        assertEquals(toUomId, conversion.toUomId());
        assertEquals(2.0, conversion.factor());
    }

    @Test
    void negativeFactor() {
        assertThrows(IllegalArgumentException.class,
                () -> new ItemUomConversion(fromUomId, toUomId, -1.0));
    }

    @Test
    void zeroFactor() {
        assertThrows(IllegalArgumentException.class,
                () -> new ItemUomConversion(fromUomId, toUomId, 0.0));
    }

    @Test
    void sameUom() {
        assertThrows(IllegalArgumentException.class,
                () -> new ItemUomConversion(fromUomId, fromUomId, 1.0));
    }

    @Test
    void nullFromUom() {
        assertThrows(NullPointerException.class,
                () -> new ItemUomConversion(null, toUomId, 1.0));
    }
}
